File: HELLO_ANDROID01/app/src/main/java/kr/co/aiai/app/MainActivity08.java
package kr.co.aiai.app;

import android.os.Bundle;
import android.util.Log;

import androidx.appcompat.app.AppCompatActivity;

public class MainActivity08 extends AppCompatActivity {

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main08);
        Log.d("gilsang","[MainActivity08]onCreate");
    }

    @Override
    protected void onDestroy() {
        super.onDestroy();
        Log.d("gilsang","[MainActivity08]onDestroy");
    }

    @Override
    protected void onStart() {
        super.onStart();
        Log.d("gilsang","[MainActivity08]onStart");
    }

    @Override
    protected void onStop() {
        super.onStop();
        Log.d("gilsang","[MainActivity08]onStop");
    }

    @Override
    protected void onResume() {
        super.onResume();
        Log.d("gilsang","[MainActivity08]onResume");
    }

    @Override
    protected void onPause() {
        super.onPause();
        Log.d("gilsang","[MainActivity08]onPause");
    }

    @Override
    protected void onRestart() {
        super.onRestart();
        Log.d("gilsang","[MainActivity08]onRestart");
    }
}

File: HELLO_AI4/ANDROID/app/src/main/java/kr/co/aiai/myapp/MainActivity08.kt
package kr.co.aiai.myapp

import android.app.AlertDialog
import android.content.Intent
import androidx.appcompat.app.AppCompatActivity
import android.os.Bundle
import android.util.Log
import android.widget.Button
import android.widget.EditText
import android.widget.TextView
import android.widget.Toast
import kotlin.random.Random

class MainActivity08 : AppCompatActivity() {

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main07)
        Log.d("gilsang","[MainActivity08]onCreate")
        var btn = findViewById<Button>(R.id.btn)
        btn.setOnClickListener() {
        }

    }

    override fun onDestroy() {
        super.onDestroy()
        Log.d("gilsang","[MainActivity08]onDestroy");
    }

    override fun onStart() {
        super.onStart()
        Log.d("gilsang","[MainActivity08]onStart");
    }

    override fun onStop() {
        super.onStop()
        Log.d("gilsang","[MainActivity08]onStop");
    }

    override fun onResume() {
        super.onResume()
        Log.d("gilsang","[MainActivity08]onResume");
    }

    override fun onPause() {
        super.onPause()
        Log.d("gilsang","[MainActivity08]onPause");
    }

    override fun onRestart() {
        super.onRestart()
        Log.d("gilsang","[MainActivity08]onRestart");
    }

}